src/niftyone/figures/generator.py

```diff
@@ -114,12 +135,35 @@ def generate(self, record: pd.Series, out_dir: Path, overwrite: bool) -> None:
 
         img_path = Path(record["finfo"]["file_path"])
         logging.info("Processing: %s", img_path)
-        img = nib.nifti1.load(img_path)
-        img = noimg.to_iso_ras(img)
+
+        signature = inspect.signature(self.view_fn)
+        view_fn_input_type = signature.parameters[
+            list(signature.parameters.keys())[0]
+        ].annotation
+
+        if get_union_subclass(view_fn_input_type, Path):
+            img = img_path
+        else:
+            img = nib.nifti1.load(img_path)
```

Comment:
Nifti specific logic in this generic method feels out of place to me.

Reply:
Yeah its teetering on the edge of being out-of-scope of the method. With the current implementation of overlays, this as a whole isn't really needed outside of dealing with diffusion data. Was more of an issue when Paths were being passed for overlaying instead of images.

Reply:
Taken out in 85e1596.